central/policy/handlers/handler.go

```diff
@@ -0,0 +1,142 @@
+package handlers
+
+import (
+	"archive/zip"
+	"context"
+	"encoding/json"
+	"net/http"
+	"strings"
+
+	"github.com/grpc-ecosystem/grpc-gateway/v2/runtime"
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/central/policy/customresource"
+	"github.com/stackrox/rox/central/policy/datastore"
+	"github.com/stackrox/rox/generated/api/v1"
+	"github.com/stackrox/rox/pkg/apiparams"
+	"github.com/stackrox/rox/pkg/httputil"
+	"github.com/stackrox/rox/pkg/jsonutil"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/set"
+	"github.com/stackrox/rox/pkg/utils"
+	"github.com/stackrox/rox/pkg/uuid"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/protoadapt"
+)
+
+var (
+	log                                      = logging.LoggerForModule()
+	maxCustomResourceNameToFilePathTruncSize = customresource.MaxCustomResourceMetadataNameLength - uuid.StringLength - 1
+)
+
+// Handler returns a handler for policy http requests
+func Handler(c datastore.DataStore) http.Handler {
+	return httpHandler{
+		policyStore: c,
+	}
+}
+
+type httpHandler struct {
+	policyStore datastore.DataStore
+}
+
+func (h httpHandler) ServeHTTP(writer http.ResponseWriter, request *http.Request) {
+	if request.Method != http.MethodPost {
+		writer.WriteHeader(http.StatusMethodNotAllowed)
+		return
+	}
+
+	var params apiparams.SaveAsCustomResourcesRequest
+	err := json.NewDecoder(request.Body).Decode(&params)
+	if err != nil {
+		httputil.WriteGRPCStyleError(writer, codes.Internal, err)
+		return
+	}
+	h.saveAsCustomResources(request.Context(), &params, writer)
+}
+
+// saveAsCustomResources saves the policies, designed by the policy ids, as custome resources
+func (h httpHandler) saveAsCustomResources(ctx context.Context, request *apiparams.SaveAsCustomResourcesRequest, writer http.ResponseWriter) {
+	policyList, missingIndices, err := h.policyStore.GetPolicies(ctx, request.IDs)
+	if err != nil {
+		httputil.WriteGRPCStyleError(writer, codes.Internal, err)
+		return
+	}
+	errDetails := &v1.PolicyOperationErrorList{}
+	for _, missingIndex := range missingIndices {
+		policyID := request.IDs[missingIndex]
+		errDetails.Errors = append(errDetails.Errors, &v1.PolicyOperationError{
+			PolicyId: policyID,
+			Error: &v1.PolicyError{
+				Error: "not found",
+			},
+		})
+		log.Debugf("A policy error occurred for id %s: not found", policyID)
+	}
+	if len(errDetails.GetErrors()) > 0 {
+		writeErrorWithDetails(writer, codes.InvalidArgument, errors.New("Failed to retrieve all policies. Check error details for a list of policies that could not be retrieved."), errDetails)
+		return
+	}
+
+	zipWriter := zip.NewWriter(writer)
+	defer utils.IgnoreError(zipWriter.Close)
+
+	writer.Header().Set("Content-Disposition", `attachment; filename="custom_resources.zip"`)
```

Comment:
const filename) up top - security_policies_<timestamp>.zip (since name of CR is SecurityPolicy - and timestamp for uniqueness? cc: @mansursyed for UX input.